apps/filters/views.py

```diff
@@ -0,0 +1,147 @@
+    )
+    data = [_to_dict(fs, request_user=request.user) for fs in qs.all()]
+    return JsonResponse({"results": data})
+
+
+@require_http_methods(["POST"])
+@login_and_team_required
+def create_filter_set(request, team_slug: str, table_type: str):
+    # Validate table_type against allowed choices
+    if not FilterSet.is_valid_table_type(table_type):
+        return JsonResponse({"error": "Invalid table_type"}, status=400)
+
+    data = {"name": request.POST.get("name"), "filter_query_string": request.POST.get("filter_query_string")}
+    serializer = FilterSetCreateUpdateSerializer(
+        data=data, context={"is_team_admin": request.team_membership.is_team_admin}
+    )
+    if not serializer.is_valid():
+        return JsonResponse(serializer.errors, status=400)
+
+    validated = serializer.validated_data
+
+    with transaction.atomic():
+        if validated.get("is_default_for_user"):
+            FilterSet.objects.filter(
+                team=request.team, user=request.user, table_type=table_type, is_default_for_user=True
+            ).update(is_default_for_user=False)
+        if validated.get("is_default_for_team"):
+            FilterSet.objects.filter(team=request.team, table_type=table_type, is_default_for_team=True).update(
+                is_default_for_team=False
+            )
+
+        try:
+            filter_set = FilterSet.objects.create(
+                team=request.team,
+                user=request.user,
+                name=validated.get("name", "").strip(),
+                table_type=table_type,
+                filter_query_string=validated.get("filter_query_string", ""),
+                is_shared=validated.get("is_shared", False),
+                is_starred=validated.get("is_starred", False),
+                is_default_for_user=validated.get("is_default_for_user", False),
+                is_default_for_team=validated.get("is_default_for_team", False),
+            )
+            return JsonResponse({"success": True, "filter_set": _to_dict(filter_set, request.user)})
+        except IntegrityError:
+            return JsonResponse({"error": "Unable to create filter set"}, status=400)
+
+
+class FilterSetView(LoginAndTeamRequiredMixin, View):
+    """Handle PATCH (edit) and DELETE operations for FilterSet objects."""
+
+    def patch(self, request, team_slug: str, pk: int):
+        """Handle PATCH request to edit a filter set."""
+        fs = get_object_or_404(FilterSet, team=request.team, id=pk, user=request.user)
+
+        payload = json.loads(request.body or b"{}")
+        serializer = FilterSetCreateUpdateSerializer(
+            data=payload, partial=True, context={"is_team_admin": request.team_membership.is_team_admin}
+        )
+        if not serializer.is_valid():
+            return JsonResponse(serializer.errors, status=400)
+        validated = serializer.validated_data
+
+        with transaction.atomic():
+            updates = []
+            if "filter_query_string" in validated:
```

Comment:
This seems like a lot of logic that could be replaced by the serializer if it was made into a ModelSerializer